### PR TITLE
Dynamo to S3 Sample DAG and Docs

### PR DESCRIPTION
**Pull request by ferruzzi**
Adds a sample DAG and docs for the existing Dynamo to S3 Transfer operator.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by ferruzzi** [comment]
@eladkal  - Another short one in the DAG/Docs improvement project if you get time, since you reviewed the previous ones.

**Comment by ferruzzi** [comment]
Failing image build; rebased

File: airflow/providers/amazon/aws/example_dags/example_dynamodb_to_s3.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    catchup=False,
+) as dag:
+
+    # [START howto_transfer_dynamodb_to_s3]
+    backup_db = DynamoDBToS3Operator(
+        task_id='backup_db',
+        dynamodb_table_name=TABLE_NAME,
+        s3_bucket_name=BUCKET_NAME,
+        # Max output file size in bytes.  If the Table is too large, multiple files will be created.
+        file_size=1000,
```
Comment (eladkal) [inline airflow/providers/amazon/aws/example_dags/example_dynamodb_to_s3.py:41]: Can you please extend the example/docs to include also the segment explanation?
https://github.com/apache/airflow/blob/602abe8394fafe7de54df7e73af56de848cdf617/airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py#L65-L87

(and remove the segment example from the comments)

**Review by eladkal — CHANGES_REQUESTED** [review]

File: airflow/providers/amazon/aws/example_dags/example_dynamodb_to_s3.py (hunk 1 above)
Comment (ferruzzi) [reply airflow/providers/amazon/aws/example_dags/example_dynamodb_to_s3.py:41]: These changed should be in https://github.com/apache/airflow/pull/21920/commits/b0278a1155942ef0fdd88b745ccdabaab4578e04

**Review by eladkal — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**